enclave/src/polkadex_balance_storage.rs

```diff
@@ -133,10 +151,58 @@ impl PolkadexBalanceStorage {
             }
         }
     }
+
+    pub fn reduce_free_balance(
+        &mut self,
+        token: AssetId,
+        acc: AccountId,
+        amt: Balance,
+    ) -> Result<(), GatewayError> {
+        match self
+            .storage
+            .get_mut(&PolkadexBalanceKey::from(token, acc).encode())
+        {
+            Some(balance) => {
+                balance.free = balance
+                    .free
+                    .checked_sub(amt)
+                    .ok_or(GatewayError::LimitOrderPriceNotFound)?; //FIXME Error type
+                Ok(())
+            }
+            None => {
+                error!("Account Id or Asset id not avalaible");
+                return Err(GatewayError::AccountIdOrAssetIdNotFound);
+            }
+        }
+    }
+
+    pub fn increase_free_balance(
+        &mut self,
+        token: AssetId,
+        acc: AccountId,
+        amt: Balance,
+    ) -> Result<(), GatewayError> {
+        match self
+            .storage
+            .get_mut(&PolkadexBalanceKey::from(token, acc).encode())
+        {
+            Some(balance) => {
+                balance.free = balance
+                    .free
+                    .checked_add(amt)
+                    .ok_or(GatewayError::LimitOrderPriceNotFound)?; //FIXME Error Type
+                Ok(())
+            }
+            None => {
+                error!("Account Id or Asset Id not available [here]");
```

Comment:
I'd suggest to create a seperate issue for that. Especially because we need to discuss how to handle new entries. For now, I'm creating new balance entries whenever there's a deposit call and no entry exists for the asset and account id (https://github.com/Polkadex-Substrate/polkadexTEE-worker/blob/feature/fm-bh-openfinex-api/enclave/src/polkadex_balance_storage.rs#L114). But we should also discuss if `set_reserve_balance` or `set_free_balance` should create a new entry instead of issuing an error..

Reply:
See issue #93